packages/passport/sdk/src/zkEvm/zkEvmProvider.ts

```diff
@@ -201,77 +131,86 @@ export class ZkEvmProvider implements Provider {
     });
   }
 
-  // Used to get the registered zkEvm address from the User session
-  async #getZkEvmAddress() {
-    try {
-      const user = await this.#authManager.getUser();
-      if (user && isZkEvmUser(user)) {
-        return user.zkEvm.ethAddress;
-      }
-      return undefined;
-    } catch {
-      return undefined;
+  // eslint-disable-next-line class-methods-use-this
+  #getZkEvmUser(user: User | null): UserZkEvm | undefined {
+    if (user && isUserZkEvm(user)) {
+      return user;
     }
+    return undefined;
   }
 
   async #performRequest(request: RequestArguments): Promise<any> {
     // This is required for sending session activity events
 
-    switch (request.method) {
-      case 'eth_requestAccounts': {
-        const zkEvmAddress = await this.#getZkEvmAddress();
-        if (zkEvmAddress) return [zkEvmAddress];
-
-        const flow = trackFlow('passport', 'ethRequestAccounts');
-
-        try {
-          const user = await this.#authManager.getUserOrLogin();
-          flow.addEvent('endGetUserOrLogin');
-
-          this.#initialiseEthSigner(user);
+    // Get user from local storage
+    const user = await this.#authManager.getUser();
+    const zkEvmUser = this.#getZkEvmUser(user);
```

Comment:
This will have unintended consequences on the passthrough methods when `getUser` is called. Two scenarios that I can think of:

1. If a token refresh is triggered, then we'll need to wait for that process to resolve before we can pass the method along to the node
2. If a token refresh is triggered and it fails, then this error will be surfaced to the caller

Reply:
>If a token refresh is triggered, then we'll need to wait for that process to resolve before we can pass the method along to the node

`await this.#authManager.getUser();` already awaits the refreshing promise so I think this is fine?

> If a token refresh is triggered and it fails, then this error will be surfaced to the caller

[Looks like this is fine](https://github.com/immutable/ts-immutable-sdk/blob/feature/ID-1897-magic-session-management/packages/passport/sdk/src/zkEvm/zkEvmProvider.ts#L245) for `eth_requestAccounts`, `personal_sign`, etc as we surface any errors anyway to the caller. 

However for `eth_accounts` and a few other RPC calls, this would throw when it should just return `[]` as we don't handle error handling. I'll give this some thought.

Reply:
https://github.com/immutable/ts-immutable-sdk/pull/1965/commits/ac1424aec118caa4427220242b68b2e06ef69c1c